firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/ProviderProxyNativeComponent.java

```diff
@@ -15,37 +15,60 @@
 package com.google.firebase.crashlytics.internal;
 
 import androidx.annotation.NonNull;
+import com.google.firebase.inject.Provider;
 import java.io.File;
 
-public final class MissingNativeComponent implements CrashlyticsNativeComponent {
+public final class ProviderProxyNativeComponent implements CrashlyticsNativeComponent {
 
   private static final NativeSessionFileProvider MISSING_NATIVE_SESSION_FILE_PROVIDER =
       new MissingNativeSessionFileProvider();
 
+  private final Provider<CrashlyticsNativeComponent> provider;
+
+  public ProviderProxyNativeComponent(Provider<CrashlyticsNativeComponent> provider) {
+    this.provider = provider;
+  }
+
   @Override
   public boolean hasCrashDataForSession(@NonNull String sessionId) {
+    if (provider.get() != null) {
+      return provider.get().hasCrashDataForSession(sessionId);
+    }
```

Comment:
I recommend caching the result of `get()` instead of calling into it twice. This will make code more readable and more conventional as per [Injecting Providers](https://github.com/google/guice/wiki/InjectingProviders#injecting-providers) where in more traditional DI setups each call to `get()` could return a newly created instance if the component is not a singleton.

While all Firebase components are singletons, it could be beneficial to be consistent with common conventions for future readers of this code(including ourselves).

```suggestion
    Foo native = provider.get();
    if (native != null) {
      return native.hasCrashDataForSession(sessionId);
    }
```

Here and throughout